src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -391,41 +434,47 @@ class CosmasGoogleCloudService(private val freeBucketName: String,
         val project = synchronized(this.fileBuffer) {
```

Comment:
What do you think of holding the lock on `fileBuffer` during the whole run of this method? It seems that we're going to have troubles if someone modifies the in-memory structure for the current project when this code is running, no?